### PR TITLE
Update CHANGELOG.md

### PR DESCRIPTION
**Pull request by daxian-dbw**
It's what Joey shared in our Team channel. I only fixed a few spelling issues.

### CODE REVIEW

**Comment by daxian-dbw** [comment]
Note that failures are expected as I updated the release versions in docs but the corresponding packages are not released yet.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,59 @@
...
+ and was only preserved in .NET Core for specific legacy purposes that are inapplicable to PowerShell.)
+ (#4027)
+
+### Engine updates and fixes
+
+- Add support to probe and load assemblies from GAC on Windows platform.
+ This means that you can now load Windows PowerShell modules with assembly dependencies which reside in the GAC.
+ If you're interested in running your traditional Windows PowerShell scripts and cmdlets using the power of .NET Standard 2.0,
+ try adding your Windows PowerShell module directories to your PowerShell Core `$PSModulePath`.
+ (E.g. `$env:PSModulePath += ';C:\Program Files\WindowsPowerShell\Modules;C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules'`)
```
Comment (SteveL-MSFT) [inline CHANGELOG.md]: Link to https://github.com/PowerShell/PowerShell/issues/4062 to get feedback and instructions to modify the psmodulepath

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,59 @@
...
+ This property was strongly tied to the Windows build version.
+ Instead, we recommend that you use `GitCommitId` to retrieve the exact build version of PowerShell Core.
+ (#3877) (Thanks to @iSazonov!)
+- Change positional parameter for `powershell.exe` from `-Command` to `-File`.
+ This fixes the usage of `#!` (aka as a shebang) in PowerShell scripts that are being executed from non-PowerShell shells on non-Windows platforms.
+ This also means that you can now do things like `powershell foo.ps1` or `powershell fooScript` without specifying `-File`.
+ However, this change now requires that you explicitly specify `-c` or `-Command` when trying to do things like `powershell.exe Get-Command`.
+ (#4019)
+- Remove `ClrVersion` property from `$PSVersionTable`.
+ (This property is largely irrelevant PowerShell Core,
```
Comment (daxian-dbw) [inline CHANGELOG.md]: Fixed. Changed to `irrelevant for .NET Core`

File: CHANGELOG.md (hunk 1 above)
Comment (daxian-dbw) [inline CHANGELOG.md]: Fixed. Changed from
```
Even if the module isn't owned by the PowerShell Team, please file issues to tell us what works and what doesn't! (#3981)
```
to
```
Even if the module isn't owned by the PowerShell Team, please tell us what works and what doesn't by leaving a comment in [issue #4062][issue-4062]! (#3981)
```

**Review by SteveL-MSFT — APPROVED** [review]
LGTM

**Review by joeyaiello — APPROVED** [review]
Next time I'll push straight here. Sorry about that! 

LGTM, though!

**Pull request merged**